### ISSUE TITLE
Add support for anniversaries 

### ISSUE THREAD

**Issue by lafrech** [Jerome-github/gbirthday#5]
Imported from https://sourceforge.net/p/gbirthday/feature-requests/2/

tellapu:

Hi!
Could it possible to integrate this idea (add support for anniversaries):
http://ubuntuforums.org/showthread.php?t=1106740&highlight=gbirthday
Thanks.

**Comment by lafrech** [comment]
Hi tellapu.

I don't understand the anniversary feature. What is the difference with the birthday feature? Would you mind elaborating, describing what you expect?

The post you link to has an attached archive I can't open unless I register to Ubuntu forums. (Besides, it is pretty old.)

**Comment by Filbuntu** [comment]
Hi Jerome!
Thanks a lot for your answer, finally I found it ;-). I thought I might get a notification.

The anniversary feature would be just an extension of the birthday feature, the anniversaries would also show in the celebration list (retrieved from TB, Evolution, list). The anniversary icons have 2 candles on the cake instead of just 1 for birthdays. This is how you tell one from the other in the main window.
slickvguy did some other changes:
3) I changed the taskbar icon's behavior. No matter how many days back or ahead you set your preferences, the icon will only 'illuminate' if you have UPCOMING birthdays or anniversaries. It'll still blink if there is either event occurring today.
4) Changed a few labels in en.lang to reflect the script mods. I only changed the labels in the en.lang (English) language file. If someone wants to update the other language files, I'm sure that would be appreciated by our non-anglo friends.
5) Fixed a few small bugs.

I can not upload the attachment for the ubuntu forum here, I will send it by email.
Thanks again for your help!

**Comment by lafrech** [comment]
Hi.

I committed the patch you sent me on [a new branch](https://github.com/Jerome-github/gbirthday/tree/dev_anniversaries). I assume the author accepts the GPLv2 license.

I had a quick look at it. It can't be merged as is for many reasons:
- It is based on a very old GBirthday release.
- It groups several independent modifications.
- Coding style...
- ...

I won't look into it anymore, but now that it is published here, anyone can try to extract useful stuff and send a PR. (I wouldn't advise to do so. You'd rather start from scratch IMHO.)

I don't think you expected me to merge it anyway. I think you mentioned it as an example of the feature you request.

I'm still not sure I get the problem.

Correct me if I'm wrong, but a birthday is just the anniversary of a birth. I can manage anniversaries with gbirthday just like I manage birthdays: set a date and a description string in my database (my .csv file).

The only issue with this is that gbirthday writes "Birthday" in a few places where you would like "Anniversary" written.

Is this the only issue you would like to be fixed or am I missing something?

If so, I think I won't do it because it is a minor cosmetic issue and I don't have time for this. And I'm not sure it's easy to do it without breaking too much stuff.

However, if someone comes up with a solution that
- answers the need
-  does not introduce too many regressions (like the need to modify current databases to adapt to the new code)
- is presented in a nice PR rebased on latest master
  I might consider it

Anyone interested, please contact me before polishing your PR, if you don't want to waste your time.

@Filbuntu, please reopen if you think I misunderstand your request, adding comprehensive description of what you expect. I most probably won't do it, but maybe someone will.

**Comment by Filbuntu** [comment]
Hi Jérôme!

Thanks a lot for looking into this, adding a new branch and responding!
Much appreciated.

Some clarifications:
1. Yes, if you use a csv-file as database it does not make a big
difference. But if you use an email client as database it makes a
difference. At least Evolution Email (maybe other clients) has a
separate field for anniversary (of a wedding date) and gbirthday may get
the date from this field (as it does for birthdays). If not, the
anniversaries won't show up in gbirthday (and you miss them).
1. If it was an anniversary, the icon showed two candles, so it could be
   easily distinguished between a birthday and anniversary. This is
   especially helpful if your database is not a csv-file where you can edit
   the description (e.g. in Evolution).

Thanks again!

**Comment by lafrech** [comment]
OK, I get it.

Apart from cosmetic considerations (different icon and strings such as "birthday today") listed above, the anniversary issue does not affect CSV and MySQL.

I believe the support for recent versions of Evolution is broken and I even removed Evolution from the backends.

I don't know if this issue is relevant to Lightning/Sunbird.

**Comment by Filbuntu** [comment]
Thanks!
I still hoped there will be a solution for Evolution again.
Not sure whether Lightening/Sunbird has an anniversary field, maybe with
an add-on.

-----Original Message-----

**Comment by lafrech** [comment]
Yeah, sorry about Evolution.

Evolution support is feasible, it just needs work. See https://github.com/Jerome-github/gbirthday/issues/1.

I don't use it, I don't even have Evolution installed. I might do it if I had time, but I don't and I won't for a while.

My priority for gbirthday would be to complete the [Qt port](https://github.com/Jerome-github/gbirthday/issues/6).

However, anyone wanting to do it is welcome to contact me. I'll be happy to integrate it.

**Comment by Filbuntu** [comment]
I understand, maybe after the Qt porting ;-). I use Evolution and still
favour it over other clients. 
Cheers
